modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java

```diff
@@ -2496,6 +2500,83 @@ public void testAddOrUpdateObjectEntry() throws Exception {
 
 	}
 
+	@FeatureFlags("LPD-21926")
+	@Test
+	public void testAddOrUpdateObjectEntryWithFriendlyURL() throws Exception {
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				true, Collections.emptyList());
+
+		ObjectEntry objectEntry1 = _addObjectEntry(
+			0, objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap());
+
+		_assertURLTitleMap(
+			HashMapBuilder.put(
+				"en_US", objectEntry1.getExternalReferenceCode()
+			).build(),
+			objectDefinition, objectEntry1);
+
+		objectEntry1 = _objectEntryLocalService.updateObjectEntry(
+			TestPropsValues.getUserId(), objectEntry1.getObjectEntryId(),
+			HashMapBuilder.<String, Serializable>put(
+				"able", "Test URL"
+			).put(
+				"externalReferenceCode", RandomTestUtil.randomString()
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+
+		_assertURLTitleMap(
+			HashMapBuilder.put(
+				"en_US", "test-url"
+			).build(),
+			objectDefinition, objectEntry1);
+
+		ObjectField objectField = ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				TestPropsValues.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap("Able")
```

Comment:
Can we replace this for this `LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())` ?

Reply:
Done!